Tests/SwiftKafkaTests/KafkaProducerTests.swift

```diff
@@ -54,120 +54,172 @@ final class KafkaProducerTests: XCTestCase {
+                        break
+                    }
+                }
+
+                XCTAssertEqual(2, acknowledgedMessages.count)
+                XCTAssertEqual(acknowledgedMessages.map(\.id).sorted(), messageIDs.sorted())
+                XCTAssertTrue(acknowledgedMessages.contains(where: { $0.topic == message1.topic }))
+                XCTAssertTrue(acknowledgedMessages.contains(where: { $0.topic == message2.topic }))
+                XCTAssertTrue(acknowledgedMessages.contains(where: { $0.key == message1.key }))
+                XCTAssertTrue(acknowledgedMessages.contains(where: { $0.key == message2.key }))
+                XCTAssertTrue(acknowledgedMessages.contains(where: { $0.value == message1.value }))
+                XCTAssertTrue(acknowledgedMessages.contains(where: { $0.value == message2.value }))
+
+                await producer.shutdownGracefully()
             }
         }
-
-        XCTAssertEqual(2, acknowledgedMessages.count)
-        XCTAssertEqual(acknowledgedMessages.map(\.id).sorted(), messageIDs.sorted())
-        XCTAssertTrue(acknowledgedMessages.contains(where: { $0.topic == message1.topic }))
-        XCTAssertTrue(acknowledgedMessages.contains(where: { $0.topic == message2.topic }))
-        XCTAssertTrue(acknowledgedMessages.contains(where: { $0.key == message1.key }))
-        XCTAssertTrue(acknowledgedMessages.contains(where: { $0.key == message2.key }))
-        XCTAssertTrue(acknowledgedMessages.contains(where: { $0.value == message1.value }))
-        XCTAssertTrue(acknowledgedMessages.contains(where: { $0.value == message2.value }))
-
-        await producer.shutdownGracefully()
     }
 
     func testProducerNotUsableAfterShutdown() async throws {
         let producer = try await KafkaProducer(config: config, logger: .kafkaTest)
         await producer.shutdownGracefully()
 
-        let message = KafkaProducerMessage(
-            topic: "test-topic",
-            value: "Hello, World!"
-        )
+        await withThrowingTaskGroup(of: Void.self) { group in
+
+            // Run Task
+            group.addTask {
+                try await producer.run()
+            }
 
-        do {
-            try await producer.sendAsync(message)
-            XCTFail("Method should have thrown error")
-        } catch {}
+            // Test Task
+            group.addTask {
+                let message = KafkaProducerMessage(
+                    topic: "test-topic",
+                    value: "Hello, World!"
+                )
+
+                do {
+                    try await producer.sendAsync(message)
+                    XCTFail("Method should have thrown error")
+                } catch {}
+
+                // This subscribes to the acknowledgements stream and immediately terminates the stream.
+                // Required to kill the run task.
+                var iterator: KafkaAsyncSequence<KafkaProducer.Acknowledgement>.AsyncIterator? = producer
```

Comment:
Since we dropping the iterator here the `didTerminate` callback on the delegate should be triggered which should lead to the `run()` task to terminate. Can we check that the `didTerminate` is triggered?

Reply:
The solution I came up with now is quite simple: I am just invoking `self.pollingSystem.terminate()` inside of `KafkaProducer.shutDownGracefully`. By doing that we don't have to worry about if the user subscribes to our `acknowledgements` `AsyncSequence` or not.